dev.py

```diff
@@ -0,0 +1,540 @@
+                        last_blip = time.time()
+                        last_log_size = log_size
+
+            ret = p.wait()
+        except:  # noqa: E722
+            p.terminate()
+            raise
+    elapsed = datetime.datetime.now() - start_time
+
+    if ret == 0:
+        print("Installation OK")
+    else:
+        if not show_build_log:
+            with open(log_filename, 'r') as f:
+                print(f.read())
+        print("Installation failed! ({0} elapsed)".format(elapsed))
+        sys.exit(1)
+    return
+
+
+def build_project(args):
+    """
+    Build a dev version of the project.
+
+    Returns
+    -------
+    site_dir
+        site-packages directory where it was installed
+
+    """
+
+    root_ok = [os.path.exists(os.path.join(ROOT_DIR, fn))
+               for fn in PROJECT_ROOT_FILES]
+    if not all(root_ok):
+        print("To build the project, run dev.py in "
+              "git checkout or unpacked source")
+        sys.exit(1)
+
+    env = dict(os.environ)
+
+    if args.debug or args.gcov:
+        # assume everyone uses gcc/gfortran
+        env['OPT'] = '-O0 -ggdb'
+        env['FOPT'] = '-O0 -ggdb'
+        if args.gcov:
+            import distutils.sysconfig
+            cvars = distutils.sysconfig.get_config_vars()
+            env['OPT'] = '-O0 -ggdb'
+            env['FOPT'] = '-O0 -ggdb'
+            env['CC'] = env.get('CC', cvars['CC']) + ' --coverage'
+            env['CXX'] = env.get('CXX', cvars['CXX']) + ' --coverage'
+            env['F77'] = 'gfortran --coverage '
+            env['F90'] = 'gfortran --coverage '
+            env['LDSHARED'] = cvars['LDSHARED'] + ' --coverage'
+            env['LDFLAGS'] = " ".join(cvars['LDSHARED'].split()[1:]) +\
+                ' --coverage'
+
+    build_dir = os.path.join(ROOT_DIR, 'build')
+
+    # Check if meson is already setup
+    if not os.path.exists(os.path.join(build_dir, 'build.ninja')):
+        setup_build(args, env)
+
+    from distutils.sysconfig import get_python_lib
+    site_dir = get_python_lib(prefix=PATH_INSTALLED, plat_specific=True)
+
+    cmd = ["ninja", "-C", "build"]
+    if args.parallel > 1:
+        cmd += ["-j", str(args.parallel)]
+
+    # Building with ninja-backend
+    ret = subprocess.call(cmd, env=env, cwd=ROOT_DIR)
+
+    if ret == 0:
+        print("Build OK")
+    else:
+        print("Build failed!")
+        sys.exit(1)
+
+    install_project(args.show_build_log)
+
+    return site_dir
+
+
+def run_mypy(args):
+    if args.no_build:
+        raise ValueError('Cannot run mypy with --no-build')
+
+    try:
+        import mypy.api
+    except ImportError as e:
+        raise RuntimeError(
+            "Mypy not found. Please install it by running "
+            "pip install -r mypy_requirements.txt from the repo root"
+        ) from e
+
+    site_dir = build_project(args)
+    config = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "mypy.ini",
+    )
```

Comment:
Here, `run_mypy` calls `build_project`, so, I haven't imported it from `runtests.py`.

Reply:
Ah, makes sense, thanks.